spec/datadog/profiling/collectors/cpu_and_wall_time_worker_spec.rb

```diff
@@ -497,6 +501,62 @@
       end
     end
 
+    context 'when heap profiling is enabled' do
+      let(:allocation_profiling_enabled) { true }
+      let(:heap_profiling_enabled) { true }
+      let(:test_num_allocated_object) { 123 }
+      # Sample all allocations to get easy-to-reason about numbers that are also not flaky in CI.
+      let(:allocation_sample_every) { 1 }
+
+      before do
+        allow(Datadog.logger).to receive(:warn)
+      end
+
+      it 'records live heap objects' do
+        pending "heap profiling isn't actually implemented just yet"
+
+        stub_const('CpuAndWallTimeWorkerSpec::TestStruct', Struct.new(:foo))
+
+        start
+
+        test_num_allocated_object.times { CpuAndWallTimeWorkerSpec::TestStruct.new }
+        allocation_line = __LINE__ - 1
+
+        cpu_and_wall_time_worker.stop
+
+        total_heap_sampled_for_test_struct = samples_for_thread(samples_from_pprof(recorder.serialize!), Thread.current)
+          .select { |s| s.locations.first.lineno == allocation_line && s.locations.first.path == __FILE__ }
+          .sum { |s| s.values[:'heap-live-samples'] }
+
+        expect(total_heap_sampled_for_test_struct).to eq test_num_allocated_object
```

Comment:
Minor: I'm curious, is the sum needed? I would expect the heap samples to aggregate to the same one sample with the total of all of them, since they have the same labels + stack. (like the allocation profiling samples do)

Reply:
Indeed! I wrote the more complicated stack_recorder version first and didn't realize the simpler context here.